clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/OperationsController.groovy

```diff
@@ -67,34 +68,39 @@ class OperationsController {
    */
   @Deprecated
   @RequestMapping(value = "/ops", method = RequestMethod.POST)
-  Map<String, String> operations(@RequestBody List<Map<String, Map>> requestBody) {
+  Map<String, String> operations(@RequestParam(value = "operationKey", required = false) String operationKey,
+                                 @RequestBody List<Map<String, Map>> requestBody) {
     List<AtomicOperation> atomicOperations = collectAtomicOperations(requestBody)
-    start(atomicOperations)
+    start(atomicOperations, operationKey)
   }
 
   /**
    * @deprecated Use /{cloudProvider}/ops/{name} instead
    */
   @Deprecated
   @RequestMapping(value = "/ops/{name}", method = RequestMethod.POST)
-  Map<String, String> operation(@PathVariable("name") String name, @RequestBody Map requestBody) {
+  Map<String, String> operation(@PathVariable("name") String name,
+                                @RequestParam(value = "operationKey", required = false) String operationKey,
+                                @RequestBody Map requestBody) {
     List<AtomicOperation> atomicOperations = collectAtomicOperations([[(name): requestBody]])
-    start(atomicOperations)
+    start(atomicOperations, operationKey)
   }
 
   @RequestMapping(value = "/{cloudProvider}/ops", method = RequestMethod.POST)
   Map<String, String> cloudProviderOperations(@PathVariable("cloudProvider") String cloudProvider,
+                                              @RequestParam(value = "operationKey", required = false) String operationKey,
                                               @RequestBody List<Map<String, Map>> requestBody) {
     List<AtomicOperation> atomicOperations = collectAtomicOperations(cloudProvider, requestBody)
-    start(atomicOperations)
+    start(atomicOperations, operationKey)
   }
 
   @RequestMapping(value = "/{cloudProvider}/ops/{name}", method = RequestMethod.POST)
   Map<String, String> cloudProviderOperation(@PathVariable("cloudProvider") String cloudProvider,
                                              @PathVariable("name") String name,
+                                             @RequestParam(value = "operationKey", required = false) String operationKey,
```

Comment:
`operationKey` vs `clientRequestId`? 

Should be consistent in naming as it's passed through methods (vs. shortening it to `key`).